### PR TITLE
Add localization guidelines

### PR DESCRIPTION
**Pull request by zacharysarah**
Add localization (l10n) to the Kubernetes contributor guidelines.

Replaces: 
- #6517 
- #6244

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/7245)

### CODE REVIEW

File: docs/home/contribute/localization.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+### Project
+
+Teams must track their overall progress with a [GitHub project](https://help.github.com/articles/creating-a-project-board/).
+
+Projects must include columns for:
+- To do
+- In progress
+- Done
+
+For example: the [Chinese localization project](For example: https://github.com/kubernetes/kubernetes-docs-cn/projects/1).
```
Comment (xiangpengzhao) [inline docs/home/contribute/localization.md]: bad format :) `For example: ` in the parentheses is redundant.

File: docs/home/contribute/localization.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+2. Select the `release-1.X` branch for the most recent version.
+
+    For example, the branch for Kubernetes v1.9 docs is `release-1.9`.
+
+Source files reside in the `/docs/` directory.
+{: .note}
+
+### Repository
+
+A l10n team will have a repository specifically dedicated to its work, for example: [kubernetes/kubernetes-docs/cn](https://github.com/kubernetes/kubernetes-docs-cn).
```
Comment (xiangpengzhao) [inline docs/home/contribute/localization.md]: `kubernetes/kubernetes-docs/cn` --> `kubernetes/kubernetes-docs-cn` ?

**Comment by xiangpengzhao** [comment]
The nits can be in a new PR :)

File: docs/home/contribute/localization.md (hunk 1 above)
Comment (zacharysarah) [reply docs/home/contribute/localization.md]: Good catch!

**Pull request merged**